latex.bzl

```diff
@@ -10,59 +12,140 @@ LatexOutputInfo = provider(
+
+    Args:
+      ctx: For accessing the inputs parameters.
+      dep_tools: the tools needed by latexrun
+      files: all files that might be needed as part of the build.
+
+    Returns:
+      A list of commands to provide to the ctx.actions.run invokation
+    """
+
+    def filter_on(files, exts):
+        directories = []
+        for inp in files:
+            dirname = inp.dirname
+            valid = False
+            if not exts:
+                valid = True
+            else:
+                for ext in exts:
+                    if inp.path.endswith(ext):
+                        valid = True
+            if valid and dirname not in directories:
+                directories.append(dirname)
+        return directories
+
+    type_env = {
+        "AFMFONTS": ".:" + ":".join(filter_on(files, [".afm"])),
+        "BIBINPUTS": ".:" + ":".join(filter_on(files, [".bib"])),
+        "ENCFONTS": ".:" + ":".join(filter_on(files, [".enc"])),
+        "LUAINPUTS": ".:" + ":".join(filter_on(files, [".lua" or ".luc"])) +
+                     ":$TEXINPUTS:",
+        "OPENTYPEFONTS": ".:" + ":".join(filter_on(files, [])),
+        "T1FONTS": ".:" + ":".join(filter_on(files, [".pfa", ".pfb"])),
+        "TEXFONTMAPS": ".:" + ":".join(filter_on(files, [".map"])) + ":",
+        "TEXINPUTS": ".:" + ":".join(filter_on(files, [])),
+        "TFMFONTS": ".:" + ":".join(filter_on(files, [".tfm"])),
+        "TTFONTS": ".:" + ":".join(filter_on(files, [".ttf", ".ttc"])),
+        "VFFONTS": ".:" + ":".join(filter_on(files, [".vf"])),
+    }
+    env = {
+        "PATH": ":".join(
+            [
+                dep_tools[0].to_list()[0].dirname,  # latex bin folder
+                dep_tools[4].to_list()[0].dirname,  # script folder
+                "/bin",  # sed, rm, etc. needed by mktexlsr
+                "/usr/bin",  # needed to find python
+            ],
+        ),
+        "SOURCE_DATE_EPOCH": "0",
+        "TEXMF": ".",
+        "TEXMFCNF": ctx.files.web2c[0].dirname,
+        "TEXMFDBS": "$PWD:$TEXMFHMNE:$TEXMF",
+        "TEXMFHOME": "$PWD",
+        "TEXMFROOT": ".",
+    }
+    env.update(type_env)
+    return env
+
+def get_engine_cmds_gen(engine_progname):
+    engine_map = {"lualatex": lualatex_engine_cmd_gen}
+    return engine_map[engine_progname]
+
 def _latex_impl(ctx):
     toolchain = ctx.toolchains["@bazel_latex//:latex_toolchain_type"].latexinfo
-    custom_dependencies = []
-    for srcs in ctx.attr.srcs:
-        for file in srcs.files.to_list():
-            if file.dirname not in custom_dependencies:
-                custom_dependencies.append(file.dirname)
-    custom_dependencies = ",".join(custom_dependencies)
-
-    flags = [
-        "--flag=--latex-args=--progname=lualatex --debug-format --output-format={} --shell-escape --jobname={}"
-            .format(ctx.attr.format, ctx.label.name),
+
+    latex_tool = getattr(toolchain, ctx.attr._engine)
+    dep_tools = [
+        toolchain.biber.files,
```

Comment:
Here we have both bib tools so it makes sense to update the api to support both with a dedicated parameter?

Reply:
done